src/data-mgt/python/batch-jobs/airqo/batch_fetch_airqo_data.py

```diff
@@ -0,0 +1,146 @@
+        url = AIRQO_API_BASE_URL + "devices/events/add?device=" + device + "&tenant=" + tenant
+
+        results = requests.post(url, json_data, headers=headers)
+
+        # print(url)
+        print(results.json())
+
+    except Exception as e:
+        print("================ Error Occurred ==================")
+        print(e)
+        traceback.print_exc()
+        print("================ Error End ==================")
+
+
+class GetDeviceMeasurements(luigi.Task):
+    """
+    Gets the device measurements
+    """
+
+    def requires(self):
+        return GetAirqoDevices()
+
+    def output(self):
+        return luigi.LocalTarget("data/device_measurements.json")
+
+    def run(self):
+
+        client = bigquery.Client()
+
+        devices = pd.read_json('data/devices.json')
+
+        details = []
+
+        # create a to hold all threads
+        threads = []
+
+        for device_index, device_row in devices.iterrows():
+
+            query = """
+            SELECT created_at as time, pm2_5, pm10 , s2_pm2_5,
+             s2_pm10, temperature as internalTemperature, humidity as internalHumidity, voltage as battery, altitude,
+             no_sats as satellites, hdope as hdop, wind as speed FROM airqo-250220.thingspeak.clean_feeds_pms WHERE
+             channel_id={0} ORDER BY created_at ASC
+               """.format(int(device_row["channelID"]))
+
+            dataframe = (
+                client.query(query).result().to_dataframe()
+            )
+
+            for index, row in dataframe.iterrows():
+
+                data = dict({
+
+                    "device": device_row['name'],
+                    "channelID": device_row["channelID"],
+
+                    "location": {"latitude": {"value": check_float(device_row["latitude"])},
+                                 "longitude": {"value": check_float(device_row["longitude"])}},
+
+                    "frequency": "minute",
+                    "time": pd.Timestamp(row["time"]).isoformat(),
+                    "pm2_5": {"value": check_float(row["pm2_5"])},
+                    "pm10": {"value": check_float(row["pm10"])},
+                    "s2_pm2_5": {"value": check_float(row["s2_pm2_5"])},
+                    "s2_pm10": {"value": check_float(row["s2_pm10"])},
+                    "battery": {"value": check_float(row["battery"])},
+                    "altitude": {"value": check_float(row["altitude"])},
+                    "speed": {"value": check_float(row["speed"])},
+                    "satellites": {"value": check_float(row["satellites"])},
+                    "hdop": {"value": check_float(row["hdop"])},
+                    "internalTemperature": {"value": check_float(row["internalTemperature"])},
+                    "internalHumidity": {"value": check_float(row["internalHumidity"])},
+                })
+
+                details.append(data)
+
+                thread = Thread(target=single_component_insertion, args=(data, "airqo"))
```

Comment:
We might need to review the naming to reflect reality, are we still doing single component insertion?